### PR TITLE
Make OGDS sync case insensitive

### PR DESCRIPTION
**Pull request by lukasgraf**
This makes the OGDS synchronisation case inensitive with regard to the user ID.

Therefore, when a user ID changes case in LDAP/AP, it will still be matched to the existing OGDS user (with original capitalization for its user ID) when updating user properties and group memberships.

For [CA-4834](https://4teamwork.atlassian.net/browse/CA-4834)

## Checklist

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

### CODE REVIEW

File: opengever/ogds/base/sync/ogds_updater.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 from sqlalchemy.sql.expression import false
 from sqlalchemy.sql.expression import or_
 from sqlalchemy.sql.expression import true
+from UserDict import UserDict
```
Comment (njohner) [inline opengever/ogds/base/sync/ogds_updater.py:25]: I did not know about that class. That's actually really useful

File: opengever/ogds/base/sync/ogds_updater.py [hunk 2]
```diff
@@ -637,22 +641,33 @@ def update_mappings(self, ldap_objects, ogds_objects, pk='userid'):
...
 
-        deleted = ogds_active_keys - ldap_keys
-        existing = ldap_keys & ogds_keys
+        deleted = [k for k in ogds_active_keys if k not in ldap_objects_ci]
+        existing = [k for k in ogds_keys if k in ldap_objects_ci]
         modified = {}
         for key in existing:
-            diff = set(ldap_objects[key].items()) ^ set(ogds_objects[key].items())
+            # Case-insensitive lookup of the matching LDAP record
+            ldap_record = ldap_objects.get(key, ldap_objects_ci[key])
```
Comment (njohner) [inline opengever/ogds/base/sync/ogds_updater.py:654]: I think you can simply use `ldap_record = ldap_objects_ci[key]` no? I don't see why you would need to first try to get from `ldap_objects`, as `ldap_objects_ci` will anyway always first look for exact matches.

File: opengever/ogds/base/sync/ogds_updater.py [hunk 3]
```diff
@@ -637,22 +641,33 @@ def update_mappings(self, ldap_objects, ogds_objects, pk='userid'):
...
+                if attributes:
+                    modified[key] = attributes
 
         added_mappings = [ldap_objects[a] for a in added_ignore_capitalization]
         deleted_mappings = [{pk: d, 'active': False} for d in deleted]
         modified_mappings = []
         for key, modified_attrs in modified.items():
             changes = {pk: key}
+            # Case-insensitive lookup of the matching LDAP record
+            ldap_record = ldap_objects.get(key, ldap_objects_ci[key])
```
Comment (njohner) [inline opengever/ogds/base/sync/ogds_updater.py:673]: Same here. `ldap_record = ldap_objects_ci[key]` should work

**Review by njohner — APPROVED** [review]
LGTM

File: opengever/ogds/base/sync/ogds_updater.py (hunk 2 above)
Comment (lukasgraf) [reply opengever/ogds/base/sync/ogds_updater.py:654]: You're right, this is a remnant of before, where this wasn't guaranteed yet.

**Pull request merged**